backend/server/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java

```diff
@@ -1876,4 +1817,29 @@ public void deleteLogVersionedInputs(Session session, List<Long> repoIds) {
     query.executeUpdate();
     LOGGER.debug("ExperimentRun versioning deleted successfully");
   }
+
+  public Future<Void> updateExperimentRunName(UpdateExperimentRunName request) {
+    final var runId = request.getId();
+    final var name = request.getName();
+    final var now = Calendar.getInstance().getTimeInMillis();
+
+    return checkPermission(
+            Collections.singletonList(runId), ModelDBActionEnum.ModelDBServiceActions.UPDATE)
+        .toFuture()
+        .thenCompose(
+            unused ->
+                jdbi.run(
+                    handle -> {
+                      String updateQueryString =
+                          "UPDATE experiment_run SET name=:name, date_updated=:date_updated WHERE id=:runId";
+                      handle
+                          .createUpdate(updateQueryString)
+                          .bind("name", name)
+                          .bind("date_updated", now)
+                          .bind("runId", runId)
+                          .execute();
+                    }))
+        .thenCompose(unused -> updateModifiedTimestampV2(runId, now))
+        .thenCompose(unused -> updateVersionNumberV2(runId));
```

Comment:
I wouldn't do this like this. Just put the updates inline into the query... Let's keep it simple unless we have some reason to think it doesn't need to be simple.

Reply:
Updated ✅